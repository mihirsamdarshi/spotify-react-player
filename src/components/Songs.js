import React, { useContext } from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Image from 'react-bootstrap/Image';
import Typography from '@material-ui/core/Typography';
import CardActionArea from '@material-ui/core/CardActionArea';
import '../stylesheets/Songs.scss';
import { GetNowPlayingDispatch, removeDuplicates, returnArtistAlbumString } from '../scripts/helpers';

const SongBox = (props) => {
    const GetNowPlaying = useContext(GetNowPlayingDispatch);

    return (
        <Card className="card" onClick={() => GetNowPlaying(props)}>
            <CardActionArea>
                <div className="songBox">
                    <Image
                        className="songCover"
                        src={props.img.url !== undefined ? props.img.url : ''}
                    />
                    <CardContent className="songContent">
                        <Typography component="h5" variant="h5" noWrap>
                            {props.name}
                        </Typography>
                        <Typography variant="subtitle1" color="textSecondary" noWrap>
                            {returnArtistAlbumString(props)}
                        </Typography>
                    </CardContent>
                </div>
            </CardActionArea>
        </Card>
    );
};

const Songs = (props) => {
    const editedSongArray = removeDuplicates(props.songs);

    return (
        <div>
            {
                editedSongArray.map((element) => (
                    <SongBox
                        key={element.track.id}
                        name={element.track.name}
                        artist={element.track.artists}
                        album={element.track.album.name}
                        img={element.track.album.images[0]}
                        isrc={element.track.external_ids.isrc}
                        uri={element.track.uri}
                    />
                ))
            }
        </div>
    );
};

export default Songs;
