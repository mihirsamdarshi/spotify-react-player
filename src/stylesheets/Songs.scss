$break-large: 1000px;

.box {
  display: flex;
  @media screen and (max-width: $break-large - 1) {
    flex-direction: column;
  }
  @media screen and (min-width: $break-large) {
    flex-direction: row;
  }
}

.content {
  display: flex;
  flex-direction: column;
  text-align: left;
  @media screen and (max-width: $break-large - 1) {
    width: 100%;
  }
  @media screen and (min-width: $break-large) {
    width: 70%;
  }
}

.cover {
  display: flex;
  @media screen and (max-width: $break-large - 1) {
    width: 100%;
  }
  @media screen and (min-width: $break-large) {
    width: 100px;
  }
  align-self: flex-start;
}
