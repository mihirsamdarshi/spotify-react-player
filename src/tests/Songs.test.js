import ReactDOM from 'react-dom';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import Songs from '../components/Songs';
import { fakeSongsData } from './utils';

const { act } = TestRenderer;

it('renders the Songs component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Songs songs={fakeSongsData} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should start with a song box', async () => {
    await act(async() => {
        const component = await TestRenderer.create(<Songs songs={fakeSongsData}/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
