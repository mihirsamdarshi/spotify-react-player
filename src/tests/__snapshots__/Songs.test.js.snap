// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should start with a song box 1`] = `
<div>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root card MuiPaper-rounded"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="songBox"
      >
        <img
          className="songCover"
          src="test.com"
        />
        <div
          className="MuiCardContent-root songContent"
        >
          <h5
            className="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap"
          >
            Test
          </h5>
          <h6
            className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-noWrap"
          >
            Testy McTestFace
          </h6>
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
  </div>
</div>
`;
